src/index.ts

```diff
@@ -51,75 +46,58 @@ class BaseClient {
       grpc.credentials.createInsecure()
     );
 
-    this.opts = opts;
-    // Use optional chaining for cleaner code
-    this.owner = opts.owner ?? undefined;
+    // this.opts = opts;
+    // this.owner = opts.owner ?? undefined;
   }
 
-  async authenticate(): Promise<void> {
```

Comment:
I think having a generic client will be error-prone. We explictly define two interfaces, auth with signature and auth with Jwt, and the application needs to understand the difference.